app/routes/codelists-management/codelist.js

```diff
@@ -0,0 +1,19 @@
+import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
+import { hash } from 'rsvp';
+
+export default class CodelistsManagementCodelistRoute extends Route {
+  @service store;
+
+  async model(params) {
+    let model = await hash({
+      codelist: this.store.findRecord('code-list', params.id),
+    });
+
+    return model;
```

Comment:
This setup looks a bit weird to me. If you use hash you could just return that directly instead of storing it in a variable first. I would either resolve the record in the variable and return a new object or remove the async and return the hash directly. Now it's a weird (imo) combination of both :smile: